From cf7491b035c7caebd872584a272b7faefef00511 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Mon, 7 Aug 2017 20:55:35 +0200
Subject: [PATCH 2/4] meson: Use configure_file() to copy typesfile

The working directory is the builddir and gnome.gtkdoc is not smart
enough to figure out whether the filename(s) passed are found in the
builddir or srcdir. Using files() doesn't help, either, since it always
generates paths to the builddir.

Other projects (GStreamer, Grilo) seem to use configure_file to copy the
typesfile to the builddir. Follow this.

This fixes the following error:

    Cannot open gdk-pixbuf.types: No such file or directory

https://bugzilla.gnome.org/show_bug.cgi?id=785964
---
 docs/reference/gdk-pixbuf/meson.build | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/docs/reference/gdk-pixbuf/meson.build b/docs/reference/gdk-pixbuf/meson.build
index 47f241094..3fb01f0e3 100644
--- a/docs/reference/gdk-pixbuf/meson.build
+++ b/docs/reference/gdk-pixbuf/meson.build
@@ -21,6 +21,12 @@ private_headers = [
 ]
 
 if get_option('with_docs')
+  # This should not be needed, but gnome.gtkdoc() does not copy the
+  # .types file into the builddir, if one is found
+  configure_file(input: 'gdk-pixbuf.types',
+                 output: 'gdk-pixbuf.types',
+                 configuration: configuration_data())
+
   gnome.gtkdoc('gdk-pixbuf',
                main_xml: 'gdk-pixbuf.xml',
                src_dir: [
-- 
2.14.1

